lib/test.js

```diff
@@ -53,6 +53,77 @@ function plan(count) {
 	this.plan(count, captureStack(this.plan));
 }
 
+function tryTest(fn) {
+	const args = [];
+	for (let i = 1, len = arguments.length; i < len; i++) {
+		args.push(arguments[i]);
+	}
+
+	let commitCalled = false;
+	let discarded = false;
+	let finished = false;
+
+	const commitFn = (ret, toCommit) => {
+		if (commitCalled) {
+			if (discarded) {
+				this.saveFirstError(new Error('The discard() was already called'));
+			} else {
+				this.saveFirstError(new Error('The commit() was already called'));
```

Comment:
It could be. I'm not sure how it would be better. I was trying to be explicit so that there is not more `commit()`/`discard()` calls than needed.

Also, while trying the solution, it was possible to have discard be implicitly default behavior. So that `t.try()` would transparently be not an assertion, until `commit()` is called. This way `discard()` is not needed at all. I'm not sure if this way is better, or if it should be more explicit like it is now.

Reply:
Discarding should be explicit. You could have a straight-forward assertion `t.true(enabled())` and then forget to commit the `t.try()` even if it failed. Then your test seems to pass even though it shouldn't have.